rasa_core/domain.py

```diff
@@ -390,33 +390,37 @@ def templates(self):
 
 
 class TemplateDomain(Domain):
+
     @classmethod
     def load(cls, filename, action_factory=None):
         if not os.path.isfile(filename):
             raise Exception(
                     "Failed to load domain specification from '{}'. "
                     "File not found!".format(os.path.abspath(filename)))
+        return cls.load_from_yaml(read_file(filename), action_factory=action_factory)
 
-        cls.validate_domain_yaml(filename)
-        data = read_yaml_file(filename)
+    @classmethod
+    def load_from_yaml(cls, yaml, action_factory=None):
+        cls.validate_domain_yaml(yaml, string_input=True)
+        data = read_yaml_string(yaml)
         utter_templates = cls.collect_templates(data.get("templates", {}))
         if not action_factory:
             action_factory = data.get("action_factory", None)
         slots = cls.collect_slots(data.get("slots", {}))
         additional_arguments = data.get("config", {})
-        return TemplateDomain(
-                data.get("intents", []),
-                data.get("entities", []),
-                slots,
-                utter_templates,
-                data.get("actions", []),
-                data.get("action_names", []),
-                action_factory,
-                **additional_arguments
+        return cls(
+            data.get("intents", []),
+            data.get("entities", []),
+            slots,
+            utter_templates,
+            data.get("actions", []),
+            data.get("action_names", []),
+            action_factory,
+            **additional_arguments
         )
 
     @classmethod
-    def validate_domain_yaml(cls, filename):
+    def validate_domain_yaml(cls, input, string_input=False):
```

Comment:
`input` needs to be renamed (shadows build in function)

do we need `string_input`? I think it is always `True`

Reply:
it wasnt in tests but i changed tests :)